File: 2021/08/23/main.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findTarget(root *TreeNode, k int) bool {
	return helper(root, root, k)
}
func helper(root, sroot *TreeNode, k int) bool {
	if root == nil {
		return false
	}

	return (2*root.Val != k && bs(sroot, k-root.Val)) || helper(root.Left, sroot, k) || helper(root.Right, sroot, k)
}
func bs(root *TreeNode, k int) bool {
	if root == nil {
		return false
	}
	if root.Val == k {
		return true
	}
	if root.Val < k {
		return bs(root.Right, k)
	}
	return bs(root.Left, k)
}

File: 2021/08/23/main.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        def helper(root, sroot, k):
            if root == None:
                return False

            return (2 * root.val != k and bs(sroot, k - root.val)) or helper(root.left, sroot, k) or helper(root.right, sroot, k)
        def bs(root, k):
            if root == None:
                return False
            if root.val == k:
                return True
            if root.val < k:
                return bs(root.right, k)
            return bs(root.left, k)
        return helper(root, root, k)